cmd/ctl/check_config.go

```diff
@@ -0,0 +1,376 @@
+	return nil
+}
+
+// verifyRouteHostMatchesCert checks OpenShift Route host in ns is present in IPS?DNS of the given TLS secret.
+func verifyRouteHostMatchesCert(ctx context.Context, kc *kube.KubernetesClient, ns string, tlsSecretName string) error {
+	// Discover route API by attempting a list via dynamic client
+	if kc.DynamicClient == nil {
+		return nil // skip
+	}
+	// Load cert IPS/DNS
+	parsed, err := x509FromTLSSecret(ctx, kc.Clientset, ns, tlsSecretName)
+	if err != nil {
+		return err
+	}
+
+	// Try routes.route.openshift.io
+	gvr := schema.GroupVersionResource{Group: "route.openshift.io", Version: "v1", Resource: "routes"}
+	routes, err := kc.DynamicClient.Resource(gvr).Namespace(ns).List(ctx, metav1.ListOptions{})
+	if err != nil || routes == nil {
+		return nil // skip if Route API not present or inaccessible
+	}
+
+	if len(routes.Items) == 0 {
+		return nil // skip if no routes exist
+	}
+
+	// Pass if any route host matches SANs
+	for _, r := range routes.Items {
+		host, _, _ := unstructured.NestedString(r.Object, "spec", "host")
+		if host == "" {
+			continue
+		}
+		if err := parsed.VerifyHostname(host); err == nil {
+			return nil
+		}
+	}
+	return fmt.Errorf("no OpenShift Route host in namespace matches TLS IPS/DNS")
+}
+
+func agentCASecretValid(ctx context.Context, kubeClient kubernetes.Interface, ns, name string) error {
+	sec, err := kubeClient.CoreV1().Secrets(ns).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if len(sec.Data) == 0 {
+		return fmt.Errorf("%s/%s: empty secret", ns, name)
+	}
+	// Expect a "ca.crt" field with PEM data
+	if _, ok := sec.Data["ca.crt"]; !ok {
+		return fmt.Errorf("%s/%s: missing ca.crt field", ns, name)
+	}
+	// Validate PEM parses into certs
+	_, err = tlsutil.X509CertPoolFromSecret(ctx, kubeClient, ns, name, "ca.crt")
+	return err
+}
+
+func clientCertNotExpired(ctx context.Context, kubeClient kubernetes.Interface, ns, name string) error {
+	parsed, err := x509FromTLSSecret(ctx, kubeClient, ns, name)
+	if err != nil {
+		return err
+	}
+	if time.Now().After(parsed.NotAfter) {
+		return fmt.Errorf("agent certificate expired at %s", parsed.NotAfter)
+	}
+	return nil
+}
+
+func clientCertSignedByPrincipalCA(ctx context.Context, agentKube kubernetes.Interface, agentNS string, principalKube kubernetes.Interface, principalNS string) error {
+	agentX509, err := x509FromTLSSecret(ctx, agentKube, agentNS, config.SecretNameAgentClientCert)
+	if err != nil {
+		return err
+	}
+	caX509, err := x509FromTLSSecret(ctx, principalKube, principalNS, config.SecretNamePrincipalCA)
+	if err != nil {
+		return err
+	}
+	if err := agentX509.CheckSignatureFrom(caX509); err != nil {
+		return fmt.Errorf("agent certificate not signed by principal CA: %w", err)
+	}
+	return nil
+}
+
+func namespaceMatchesAgentSubject(ctx context.Context, agentKube kubernetes.Interface, agentNS string, principalKube kubernetes.Interface, principalNS string) error {
```

Comment:
VS Code is showing a warning for this function:
`unused parameter: principalNS`